install.py

```diff
@@ -93,11 +93,36 @@ def check_distribution_compatibility():
 	else:
 		log("Sorry, the installer doesn't support {0}. Aborting installation!".format(dist_name), level=2)
 
+def import_with_install(package):
+	# copied from https://discuss.erpnext.com/u/nikunj_patel
+	# https://discuss.erpnext.com/t/easy-install-setup-guide-for-erpnext-installation-on-ubuntu-20-04-lts-with-some-modification-of-course/62375/5
 
-def get_distribution_info():
+	# need to move to top said v13 for fully python3 era
+	import importlib
+
+	try:
+		importlib.import_module(package)
+	except ImportError:
+		# caveat : pip3 must be installed
+
+		import pip
+
+		pip.main(['install', package])
+	finally:
+		globals()[package] = importlib.import_module(package)
+
+
+def get_distribution_info(args):
 	# return distribution name and major version
 	if platform.system() == "Linux":
-		current_dist = platform.dist()
+		if args.python_version == "3":
+			install_package('pip3', 'python3-pip')
+
+			import_with_install('distro')
+
+			current_dist = distro.linux_distribution(full_distribution_name=True)
+		else:
+			current_dist = platform.dist()
```

Comment:
`platform.dist` is functional until `PY3.7` and is only deprecated in `PY3.8`. Check the minor version too, and install distro only if `PY3.8`+.

Reply:
using sys.version_info and only > 3.7 to install distro